### PR TITLE
Add Dump and Loader

### PR DESCRIPTION
**Pull request by greatroar**
TODO:
* [x] Tests, including a fuzz test.
* [x] Can we skip the check for 2³² blocks by storing len(f.b)-1?
* [x] Documentation, including of the file format.
* [x] Handle SyncFilter.

### CODE REVIEW

File: io.go [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+	}
+
+	version := binary.LittleEndian.Uint32(l.buf[8:])
+	l.nblocks = 1 + uint64(binary.LittleEndian.Uint32(l.buf[12:]))
+	l.nhashes = int(binary.LittleEndian.Uint32(l.buf[16:]))
+	comment := l.buf[20:]
+
+	switch {
+	case string(l.buf[:8]) != "blobloom":
+		err = errors.New("blobloom: not a Bloom filter dump")
```
Comment (ankit-bagdeZ) [inline io.go:113]: No corresponding dump is present for this.
Thus, it's failing. Can you please check this @greatroar

Comment (greatroar) [reply io.go:111]: Fixed and tested, thanks. I cherry-picked your Equals method onto master as preparation for this PR.

**Comment by ankit-bagdeZ** [comment]
Thanks, @greatroar. This works, can you please merge this?

**Comment by ankit-bagdeZ** [comment]
@greatroar can you please add the `Equals` function for `syncFilter` too?
And make a release.
Thanks!

**Comment by greatroar** [comment]
What do you want to use SyncFilter.Equals for, aside from testing?

**Comment by ankit-bagdeZ** [comment]
Yes, for validation and testing

**Comment by greatroar** [comment]
I'd rather not add hard-to-use public methods that need documentation on how to use them safely (with separate synchronization constructs).

I'll release once I'm confident that I got the right API. Until then, please use the commit SHA-1 in your go.mod file.

**Pull request merged**